src/catnip/runtime/memory/mbuf.rs

```diff
@@ -114,9 +116,7 @@ impl Drop for Mbuf {
         if self.ptr.is_null() {
             return;
         }
-        unsafe {
-            rte_pktmbuf_free(self.ptr);
-        }
+        MemoryPool::free_mbuf(self.ptr);
         self.ptr = ptr::null_mut();
```

Comment:
This seems very un-Rustian.  Now that the unsafe code is contained inside MemoryPool::free_mbuf(), it seems strange to be nullifying self.ptr here.  I would expect free_mbuf() to be doing this for us, so we don't have access to a bogus pointer upon return.

Reply:
Somehow we should make the raw pointer null, because it is not managed by Rust. I did not get what you are suggesting. Could you clarify it further?

Reply:
I was suggesting that free_mbuf() null out self.ptr itself.  To really do this right, we (or rather, the MemoryPool) would hide the "*mut rte_mbuf" inside a Rust type of our own creation.  We'd pass that type into free_mbuf(), it would take ownership, etc.

But don't worry about it, this is probably good enough for now.

Reply:
Oh, I got it and agree. I'm refactoring `Mbuf` so as to make this interface cleaner too. Thanks.